model/dependencies.go

```diff
@@ -42,14 +43,22 @@ func (di *dependencyIncluder) include(initialDeps []TVPair) ([]TVPair, error) {
 			outPairs = append(outPairs, pair)
 		}
 	}
+	// we deactivate variants that do not have specific activation set in the
+	// generated project but have been generated as dependencies of other
+	// explicitly inactive variants
+	for variant, addToActivationInfo := range di.deactivateGeneratedVariantTasks {
```

Comment:
I'm slightly confused by this, but I'd like to understand. The comment on the `activationVariants` field says that it's all variants that have some specific activation, but the comment around `pairSpecifiesActivation` implies that we should deactivate those variants where `di.deactivateGeneratedVariantTasks = true`. Why does this append variants that are meant to be deactivated?

Reply:
That is the way that specific activation is computed (checking if activation is non nil amongst other things) but the way it is used is looking like variants / tasks with "specific activation" are marked as inactive no matter what the value actually is. This is a bug since setting `activate: true` in the generated project will actually cause the task/variant to be deactivated. I was thinking we should start referring to it explicitly as "deactivation" rather than "specific activation" in the variable function naming, but open to other suggestions.

Reply:
For the purpose of not expanding this PR I'll ticket it out: https://jira.mongodb.org/browse/EVG-18543